core/vcpu.c

```diff
@@ -4311,6 +4316,66 @@ static void vcpu_state_dump(struct vcpu_t *vcpu)
             vcpu->state->_idt.base,
             vcpu->state->_idt.limit,
             (uint32_t)vcpu->state->_efer);
+
+    // Dump FPU state
+    hax_log(HAX_LOGW,
+            "FX layout:\n"
+            "fcw:    %08x  fsw:    %08x  ftw: %08x  res1: %08x  fop: %08x\n"
+            "fpu_ip: %08llx  fpu_dp: %08llx\n"
+            "mxcsr:  %08x  mxcsr_mask: %08x\n",
+            gfx->fcw, gfx->fsw, gfx->ftw, gfx->res1, gfx->fop, gfx->fpu_ip,
+            gfx->fpu_dp, gfx->mxcsr, gfx->mxcsr_mask);
+
+    for (i = 0; i < 16; i++) {
+        hax_log(HAX_LOGW, "st_mm[%d] %02x%02x%02x%02x%02x%02x%02x%02x\n", i,
+                gfx->st_mm[0][i], gfx->st_mm[1][i], gfx->st_mm[2][i],
+                gfx->st_mm[3][i], gfx->st_mm[4][i], gfx->st_mm[5][i],
+                gfx->st_mm[6][i], gfx->st_mm[7][i]);
+    }
+
+    for (i = 0; i < 16; i++) {
+        hax_log(HAX_LOGW, "mmx_1[%d] %02x%02x%02x%02x%02x%02x%02x%02x\n", i,
+                gfx->mmx_1[0][i], gfx->mmx_1[1][i], gfx->mmx_1[2][i],
+                gfx->mmx_1[3][i], gfx->mmx_1[4][i], gfx->mmx_1[5][i],
+                gfx->mmx_1[6][i], gfx->mmx_1[7][i]);
+    }
+
+    for (i = 0; i < 16; i++) {
+        hax_log(HAX_LOGW, "mmx_2[%d] %02x%02x%02x%02x%02x%02x%02x%02x\n", i,
+                gfx->mmx_2[0][i], gfx->mmx_2[1][i], gfx->mmx_2[2][i],
+                gfx->mmx_2[3][i], gfx->mmx_2[4][i], gfx->mmx_2[5][i],
+                gfx->mmx_2[6][i], gfx->mmx_2[7][i]);
+    }
+
+    for (i = 0; i < 96; i += 8) {
+        hax_log(HAX_LOGW,"pad %02x%02x%02x%02x%02x%02x%02x%02x\n",
```

Comment:
Add the colon and spaces for format.

    hax_log(HAX_LOGW, "pad: %02x%02x%02x%02x %02x%02x%02x%02x\n",

Reply:
OK